CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.0)
+project(dmon VERSION 0.0.1 LANGUAGES C)
+
+set(CMAKE_C_STANDARD 90)
+cmake_policy(SET CMP0063 NEW)
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+
+# control where the static and shared libraries are built so that on windows
+# we don't need to tinker with the path to run the executable
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}")
+
+add_library("${PROJECT_NAME}_compiler_flags" INTERFACE)
+target_compile_features("${PROJECT_NAME}_compiler_flags" INTERFACE "c_std_${CMAKE_C_STANDARD}")
+
+set(gcc_like "$<COMPILE_LANG_AND_ID:C,CXX,ARMClang,AppleClang,Clang,GNU,LCC>")
+set(msvc "$<COMPILE_LANG_AND_ID:C,CXX,MSVC>")
+# -g -fsanitize=address -fno-omit-frame-pointer
+target_compile_options(
+        "${PROJECT_NAME}_compiler_flags"
+        INTERFACE
+        "$<${gcc_like}:$<BUILD_INTERFACE:-Wall;-Wextra;-Wshadow;-Wformat=2;-Wunused;-pedantic->>"
+        "$<${msvc}:$<BUILD_INTERFACE:-W3;-WX;-Zi;-permissive->>"
+)
+if (CMAKE_C_COMPILER_ID STREQUAL "Clang" OR CMAKE_C_COMPILER_ID STREQUAL "AppleClang")
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+    set(CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+endif()
+
+# configure a header file to pass the version number only
+configure_file(
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/config.h.in"
+        "${PROJECT_NAME}Config.h"
+)
+
+add_subdirectory("src")
+
+option(BUILD_TESTS "Build tests" ON)
+if (BUILD_TESTS)
+    include(CTest)
+    enable_testing()
```

Comment:
`include(CTest)` already have a option var `BUILD_TESTING`. 
so you just need to specify `include(CTest)` e.g. after the `project(...)` call. 
Therefore you can remove `enable_testing`, `option(BUILD_TESTS "Build tests" ON)` 
and replace `if (BUILD_TESTS)` with `if (BUILD_TESTING)`

If you want to make users, which don't know that `BUILD_TESTING` is available, you could add either `ption(BUILD_TESTING "Build tests" ON)` before `include(CTest)` or adding https://cmake.org/cmake/help/latest/module/FeatureSummary.html and add `BUILD_TESTING` to the feature set. 

See https://cmake.org/cmake/help/latest/module/CTest.html#module:CTest